### PR TITLE
 Add design doc for Python APIs

### LINKED ISSUE #209: Update Autotune design doc for ML APIs
Update Autotune design doc for ML APIs

### PR DESCRIPTION
**Pull request by agarwalrounak**
Closes #209

### CODE REVIEW

File: design/python-api.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    }
+    ```
+    
+    Response:
+    
+    | Status code | Response body |
+    | --- | --- |
+    | 200 | trial_number |
+    | 400 | 1 |
+    | 403 | 1 |
```
Comment (chandrams) [inline design/python-api.md:31]: Is it 1 or -1?

File: design/python-api.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+-   Get trial JSON object.
+    
+    `GET /experiment_trials?id={sha256sum-of-autotune-object}&trial_number={trial-number}`
+    
+    Example Request:
+    
+    `curl -H 'Accept: application/json' http://<URL>:<PORT>/experiment_trials?id='sha256sum'&trial_number=0`
```
Comment (chandrams) [inline design/python-api.md:68]: This needs to be changed to quote the entire http command instead of just id

File: design/python-api.md (hunk 1 above)
Comment (agarwalrounak) [reply design/python-api.md]: It should be -1. Thanks for pointing it out. I have made the changes.

File: design/python-api.md (hunk 2 above)
Comment (agarwalrounak) [reply design/python-api.md]: Done.

**Comment by dinogun** [comment]
Closing this issue as these changes are now merged in the kruize/hpo repo through this [PR](https://github.com/kruize/hpo/pull/42).

**Pull request closed without merging**